bento/geometry/_geometry.py

```diff
@@ -1,310 +1,171 @@
+            Spatially formatted SpatialData
+        shape_names : str or list of str
+            Names of the shapes to add.
+
+        Returns
+        -------
+    sdata : spatialdata.SpatialData
+            .shapes['cell_boundaries'][shape_name]
     """
-    adata = data.copy() if copy else data
-
-    if len(xlims) < 1 and len(xlims) > 2:
-        return ValueError("Invalid xlims")
-
-    if len(ylims) < 1 and len(ylims) > 2:
-        return ValueError("Invalid ylims")
-
-    xmin, xmax = xlims[0], xlims[1]
-    ymin, ymax = ylims[0], ylims[1]
-    box = Polygon([[xmin, ymin], [xmin, ymax], [xmax, ymax], [xmax, ymin]])
-    in_crop = get_shape(data, "cell_shape").within(box)
-
-    adata = data[in_crop, :]
-    sync(adata, copy=False)
-
-    return adata if copy else None
 
+    # Cast to list if not already
+    if isinstance(shape_names, str):
+        shape_names = [shape_names]
 
-def get_shape(adata: AnnData, shape_name: str) -> gpd.GeoSeries:
-    """Get a GeoSeries of Polygon objects from an AnnData object.
+    shapes_found = set(shape_names).intersection(set(sdata.shapes[cell_shape].columns.tolist()))
+    if shapes_found == set(shape_names):
+        return sdata
+    
+    shape_names = list(set(shape_names).difference(set(sdata.shapes[cell_shape].columns.tolist())))
+    sjoined_shapes = sdata.shapes[cell_shape].copy()
+    for shape in shape_names:
+        sjoined_shapes = sjoined_shapes.sjoin(sdata.shapes[shape], how='left', predicate='contains')
+        sjoined_shapes.rename(columns={"index_right": shape}, inplace=True)
+        for index in list(sjoined_shapes.index):
+            shape_id = sjoined_shapes.at[index, shape]
+            try:
+                sjoined_shapes.at[index, shape] = sdata.shapes[shape].loc[shape_id][shape]
+            except:
+                pass
+            
+    sdata.shapes[cell_shape] = sjoined_shapes
+
+def get_shape(sdata: SpatialData, shape_name: str, sync: bool = False) -> gpd.GeoSeries:
+    """Get a GeoSeries of Polygon objects from an SpatialData object.
 
     Parameters
     ----------
-    adata : AnnData
-        Spatial formatted AnnData object
+    sdata : SpatialData
+        Spatial formatted SpatialData object
     shape_name : str
-        Name of shape column in adata.obs
-
+        Name of shape column in sdata.shapes
+    sync : bool
+        Whether to sync the shape to cell_boundaries. Default False.
+        
     Returns
     -------
     GeoSeries
         GeoSeries of Polygon objects
     """
-    if shape_name not in adata.obs.columns:
-        raise ValueError(f"Shape {shape_name} not found in adata.obs.")
-
-    if adata.obs[shape_name].astype(str).str.startswith("POLYGON").any():
-        return gpd.GeoSeries(
-            adata.obs[shape_name]
-            .astype(str)
-            .apply(lambda val: wkt.loads(val) if val != "None" else None)
-        )
-
-    else:
-        return gpd.GeoSeries(adata.obs[shape_name])
-
-
-def rename_shapes(
```

Comment:
still need this function :)

see use case:
https://github.com/ckmah/bento-manuscript/blob/main/notebooks/figure-analysis/Cardiomyocytes_analysis.ipynb